### DISCUSSION TITLE
Saving a  user defined layout / switching between user defined layouts

### DISCUSSION THREAD

**Discussion by algotron** [pthom/hello_imgui#113] [Q&A]
I start with a layout, and move the original windows around, hiding some windows. I want to save that user-changed layout as a new layout that I can switch between. I actually want to be able to save multiple user created layouts and switch between them. How do I create a new layout from the existing layout as it appears to the user on the screen?

My efforts:
------------

After I have interactively split the windows by dragging them into new positions, I have tried to save the new layout from
HelloImGui::GetRunnerParams()->dockingParams
into
HelloImGui::GetRunnerParams()->alternativeDockingLayouts

but the docking splits are not saved. When new splits are created interactively, those splits do not appear in the
HelloImGui::GetRunnerParams()->dockingParams.dockingSplits

....but this does not work. The user modified layout is not saved.

**Comment by pthom** [comment]
I'm sorry there is no easy way to do that since ImGui simply does not want to transmit this information. You will have to do it by code.

Another possibility:
Provide several identical layouts "CustomLayout1, CustomLayout2, etc" (identical as far as the code is concerned) that the user can rearrange.separately . The user modifications will be saved for each of them.

**Comment by algotron** [comment]
Thank-you Pascal,

It would be great if someday it could happen. 

Yours truly,

Peter Walker
